src/Paramore.Brighter/Transforms/Transformers/CloudEventsTransformer.cs

```diff
@@ -84,9 +97,160 @@ public Message Wrap(Message message, Publication publication)
         message.Header.Subject = _subject ?? publication.Subject;
         message.Header.ContentType = _dataContentType ?? publication.ContentType;
         message.Header.SpecVersion = _specVersion ?? message.Header.SpecVersion;
+
+        if (_format == CloudEventFormat.Binary)
+        {
+            return message;
+        }
+
+        var cloudEvent = new CloudEventMessage
+        {
+            Id = message.Id,
+            SpecVersion = message.Header.SpecVersion,
+            Source = message.Header.Source,
+            Type = message.Header.Type,
+            DataContentType = message.Header.ContentType,
+            DataSchema = message.Header.DataSchema,
+            Subject = message.Header.Subject,
+            Time = message.Header.TimeStamp,
+            Data = JsonSerializer.Deserialize<JsonElement>(message.Body.Value)
+        };
+
+        message.Body = new MessageBody(JsonSerializer.Serialize(cloudEvent, JsonSerialisationOptions.Options));
+        message.Header.ContentType = "application/cloudevents+json";
+
         return message;
     }
 
+    /// <inheritdoc />
     public Message Unwrap(Message message)
-        => message;
+    {
+        if (_format == CloudEventFormat.Binary)
+        {
+            return message;
+        }
+
+        var cloudEvents = JsonSerializer.Deserialize<CloudEventMessage>(message.Body.Value, JsonSerialisationOptions.Options);
+
+        if (cloudEvents == null)
+        {
+            return message;
+        }
+
+        var bag = cloudEvents.AdditionalProperties ?? [];
```

Comment:
What do we do if names collide? i.e., can you overwrite an existing header? Document if we think you should - this is an override - that is the behaviour, otherwise check not contains

Reply:
`System.Text.Json` will override the value, I'll document it